lib/irmin.mli

```diff
@@ -1003,6 +1003,28 @@ module Ref: sig
 
 end
 
+(** [Metadata] defines metadata that is attached to contents but stored in
+    nodes. The Git backend uses this to indicate the type of file (normal,
+    executable or symlink). *)
+module Metadata: sig
+
+  module type S = sig
+
+    include Ir_hum.S
```

Comment:
Don't use private `Ir_*` interface in the public interface. That could break linking.